src/libs/kata-types/src/config/hypervisor/dragonball.rs

```diff
@@ -0,0 +1,185 @@
+
+/// Configuration information for dragonball.
+#[derive(Default, Debug)]
+pub struct DragonballConfig {}
+
+impl DragonballConfig {
+    /// Create a new instance of `DragonballConfig`.
+    pub fn new() -> Self {
+        DragonballConfig {}
+    }
+
+    /// Register the dragonball plugin.
+    pub fn register(self) {
+        let plugin = Arc::new(self);
+        register_hypervisor_plugin(HYPERVISOR_NAME_DRAGONBALL, plugin);
+    }
+}
+
+impl ConfigPlugin for DragonballConfig {
+    fn get_max_cpus(&self) -> u32 {
+        MAX_DB_VCPUS
+    }
+    fn get_min_memory(&self) -> u32 {
+        MIN_DB_MEMORY_SIZE
+    }
+    fn name(&self) -> &str {
+        HYPERVISOR_NAME_DRAGONBALL
+    }
+
+    /// Adjust the configuration information after loading from configuration file.
+    fn adjust_configuration(&self, conf: &mut TomlConfig) -> Result<()> {
+        if let Some(db) = conf.hypervisor.get_mut(HYPERVISOR_NAME_DRAGONBALL) {
+            resolve_path!(db.jailer_path, "Dragonball jailer path {} is invalid: {}")?;
+
+            if db.boot_info.kernel.is_empty() {
+                db.boot_info.kernel = default::DEFAULT_DB_GUEST_KERNEL_IMAGE.to_string();
+            }
+            if db.boot_info.kernel_params.is_empty() {
+                db.boot_info.kernel_params = default::DEFAULT_DB_GUEST_KERNEL_PARAMS.to_string();
+            }
+
+            if db.cpu_info.default_maxvcpus > default::MAX_DB_VCPUS {
+                db.cpu_info.default_maxvcpus = default::MAX_DB_VCPUS;
+            }
+
+            if db.machine_info.entropy_source.is_empty() {
+                db.machine_info.entropy_source = default::DEFAULT_DB_ENTROPY_SOURCE.to_string();
+            }
+
+            if db.memory_info.default_memory == 0 {
+                db.memory_info.default_memory = default::DEFAULT_DB_MEMORY_SIZE;
+            }
+            if db.memory_info.memory_slots == 0 {
+                db.memory_info.memory_slots = default::DEFAULT_DB_MEMORY_SLOTS;
+            }
+        }
+        Ok(())
+    }
+
+    /// Validate the configuration information.
+    fn validate(&self, conf: &TomlConfig) -> Result<()> {
+        if let Some(db) = conf.hypervisor.get(HYPERVISOR_NAME_DRAGONBALL) {
+            if !db.path.is_empty() {
+                return Err(eother!("Path for dragonball hypervisor should be empty"));
+            }
+            if !db.valid_hypervisor_paths.is_empty() {
+                return Err(eother!(
+                    "Valid hypervisor path for dragonball hypervisor should be empty"
+                ));
+            }
+            if !db.ctlpath.is_empty() {
+                return Err(eother!("CtlPath for dragonball hypervisor should be empty"));
+            }
+            if !db.valid_ctlpaths.is_empty() {
+                return Err(eother!("CtlPath for dragonball hypervisor should be empty"));
+            }
+            validate_path!(db.jailer_path, "Dragonball jailer path {} is invalid: {}")?;
```

Comment:
Consistency: "dragonball" or "Dragonball"? Both are used on this PR but it would be good to standardise on a single capitalisation format (and potentially [adding it to the spell check dictionary](https://github.com/kata-containers/tests/tree/main/cmd/check-spelling/#adding-a-new-word)).

Reply:
fixed